vital/types/rotation.cxx

```diff
@@ -50,22 +50,20 @@ template < typename T >
 rotation_< T >
 ::rotation_( const T& yaw, const T& pitch, const T& roll )
 {
-  using std::cos;
-  using std::sin;
-  static const double root_two = std::sqrt( static_cast<double>(2.0) );
-  static const T inv_root_two = static_cast< T > ( 1.0 / root_two );
-
-  // compute the rotation from North-East-Down (NED) coordinates to
-  // East-North-Up coordinates (ENU). It is a 180 degree rotation about
-  // the axis [1/sqrt(2), 1/sqrt(2), 0]
-  const rotation_< T > Rned2enu( Eigen::Quaternion< T > ( 0, inv_root_two, inv_root_two, 0 ) );
-  const double half_x = 0.5 * static_cast< double > ( -roll );
-  const double half_y = 0.5 * static_cast< double > ( -pitch );
-  const double half_z = 0.5 * static_cast< double > ( -yaw );
-  rotation_< T > Rx( Eigen::Quaternion< T > ( T( cos( half_x ) ), T( sin( half_x ) ), 0, 0 ) );
-  rotation_< T > Ry( Eigen::Quaternion< T > ( T( cos( half_y ) ), 0, T( sin( half_y ) ), 0 ) );
-  rotation_< T > Rz( Eigen::Quaternion< T > ( T( cos( half_z ) ), 0, 0, T( sin( half_z ) ) ) );
-  *this = Rx * Ry * Rz * Rned2enu;
+  T const half_x = T( 0.5 ) * roll;
```

Comment:
```suggestion
  T const half_x = T{ 0.5 } * roll;
```
If you want to prohibit narrowing casts

Reply:
In this case I don't think we do? Consider when `T` is `float` - we would want the `0.5` to be narrowed to `float`, right?

Reply:
Actually, this is probably better written:
```suggestion
  T const half_x = static_cast< T >( 0.5 ) * roll;
```
(Likewise below...)

Reply:
I think my concern was if `T` was not a `double` or `float` and instead was an integer type. But that's probably pedantic and wouldn't happen in practice.